### PR TITLE
feat: create auto refresh

### PR DESCRIPTION
**Pull request by esoubiran-aneo**
- Disabled by default

### CODE REVIEW

File: apps/app/src/app/pages/pages/sessions/pages/session-detail/session-detail.component.ts [hunk 1]
```diff
@@ -33,11 +35,15 @@ export class SessionDetailComponent implements OnInit {
...
-    private tasksService: TasksService
+    private tasksService: TasksService,
+    public autoRefreshService: AutoRefreshService
   ) {
     this.browserTitleService.setTitle(
       this.languageService.instant('pages.sessions.session-detail.title')
     );
+
+    // Activate auto refresh
+    this.autoRefreshService.setFn(() => this.onManualTasksRefresh());
```
Comment (lemaitre-aneo) [inline apps/app/src/app/pages/pages/sessions/pages/session-detail/session-detail.component.ts:46]: call directly `onTaskRefresh` ?

`onTaskRefresh`should most likely have an optional argument ( or nullable).

Make it consistent with `apps/app/src/app/pages/pages/sessions/sessions.component.ts`

File: apps/app/src/app/shared/components/timer-interval-selector/timer-interval-selector.component.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  </button>
+  <clr-dropdown-menu clrPosition="bottom-left" *clrIfOpen>
+    <button
+      type="button"
+      clrDropdownItem
+      *ngFor="let timerItem of timersList"
+      (click)="onClick(timerItem)"
+      [clrDisabled]="timerItem === timer"
+    >
+      {{ timerItem / 1_000 }}s
```
Comment (lemaitre-aneo) [inline apps/app/src/app/shared/components/timer-interval-selector/timer-interval-selector.component.html:14]: Could could be nice to have a unit formatting system. Not for this PR.

File: apps/app/src/app/shared/services/auto-refresh.service.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  setTimer(timer: number) {
+    this.timer = timer;
+    return this;
+  }
+
+  /**
+   * Enable auto refresh
+   */
+  enable() {
+    if (this.fn) {
```
Comment (lemaitre-aneo) [inline apps/app/src/app/shared/services/auto-refresh.service.ts:43]: if is always true. Couldn't it be better to have a nullable `fn`?

File: apps/app/src/app/shared/services/auto-refresh.service.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      this.disable();
+    } else {
+      this.enable();
+    }
+  }
+
+  /**
+   * Restart the auto refresh
+   */
+  restart() {
```
Comment (lemaitre-aneo) [inline apps/app/src/app/shared/services/auto-refresh.service.ts:72]: should most likely be integrated directly into `setTimer`, and removed.

**Review by lemaitre-aneo — CHANGES_REQUESTED** [review]

File: apps/app/src/app/shared/services/auto-refresh.service.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  /**
+   * Set timer to be used for auto refresh
+   *
+   * @param timer
+   *
+   * @returns this
+   */
+  setTimer(timer: number) {
```
[the remark this answers is not in the document]
Comment (esoubiran-aneo) [reply apps/app/src/app/shared/services/auto-refresh.service.ts:34]: done

File: apps/app/src/app/shared/services/auto-refresh.service.ts (hunk 4 above)
Comment (esoubiran-aneo) [reply apps/app/src/app/shared/services/auto-refresh.service.ts:72]: done

File: apps/app/src/app/shared/services/auto-refresh.service.ts (hunk 3 above)
Comment (esoubiran-aneo) [reply apps/app/src/app/shared/services/auto-refresh.service.ts:43]: remoe if and an error is thrown only if dev forget to set fn

File: apps/app/src/app/pages/pages/sessions/pages/session-detail/session-detail.component.ts (hunk 1 above)
Comment (esoubiran-aneo) [reply apps/app/src/app/pages/pages/sessions/pages/session-detail/session-detail.component.ts:46]: done

**Review by lemaitre-aneo — DISMISSED** [review]

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**